crates/crypto/src/hash/mod.rs

```diff
@@ -39,12 +41,32 @@ impl FromStr for HashAlgorithm {
     }
 }
 
-pub trait SupportedDigest: Digest + private::Sealed {
+static EMPTY_HASH_CACHE: Lazy<HashMap<u8, Hash<Sha256>>> = Lazy::new(|| {
+    let mut m: HashMap<u8, Hash<Sha256>> = HashMap::new();
+    fn empty_hash<D: SupportedDigest>(m: &mut HashMap<u8, Hash<D>>, n: u8) -> Hash<D> {
+        let hash: Hash<D> = if n == 0 {
+            Hash::of("")
+        } else {
+            let last_hash = empty_hash(m, n - 1);
+            Hash::of(Hash::<Sha256>::of((&last_hash, &last_hash)))
```

Comment:
The logic for branch and leaf hashes should be updated in [`map.rs`](https://github.com/bytecodealliance/registry/blob/main/crates/transparency/src/map/map.rs#L241) and the empty value hash should be added there too.